utils/results/results.go

```diff
@@ -35,6 +37,33 @@ type SecurityCommandResults struct {
 	errorsMutex  sync.Mutex `json:"-"`
 }
 
+// We have three types of results: vulnerabilities, violations and licenses.
+// If the user provides a violation context (watches, repo_path, project_key, git_repo_key) the results will only include violations.
+// If the user provides a violation context and requests vulnerabilities, the results will include both vulnerabilities and violations.
+// If the user doesn't provide a violation context, the results will include vulnerabilities.
+// Only one (Resource) field can be provided at a time.
+// License information can be provided in all cases if requested.
+type ResultContext struct {
+	// If watches are provided, the scan will be performed only with the provided watches.
+	Watches []string `json:"watches,omitempty"`
+	// (Resource) If repo_path is provided, the scan will be performed on the repository's project watches.
+	RepoPath string `json:"repo_path,omitempty"`
+	// (Resource) If projectKey is provided we will fetch the watches defined on the project.
+	ProjectKey string `json:"project_key,omitempty"`
+	// (Resource) If gitRepository is provided we will fetch the watches defined on the git repository.
+	GitRepoHttpsCloneUrl string `json:"git_repo_key,omitempty"`
+	// If non of the above is provided or requested, the results will include vulnerabilities
+	IncludeVulnerabilities bool `json:"include_vulnerabilities"`
+	// If requested, the results will include licenses
+	IncludeLicenses bool `json:"include_licenses"`
+	// The active watches defined on the resources above that is fetched from the platform
```

Comment:
I think the comment should be:
The active watches defined on the project_key and git_repository values above that were fetched from the platform